### PR TITLE
Add compose util

### LINKED ISSUE #1: Compose custom CSS properties
Hey @hauptrolle thanks for this library, this is really useful! 

What do you think of adding a `compose` function (or something with a better name 😅) to create new utils based on specific CSS properties?

Was thinking something like:

```ts
function compose(...properties: (keyof CSSStyleDeclaration)[]) {
    return (config: never) => (value: string) => properties.reduce((final, cssProp) => {
        return {
            ...final,
            [cssProp]: value
        }
    }, {});
}
```

which would then be used like:

```js
export const { styled, css } = createStyled({
  utils: {
    size: compose('width','height'),
  },
});
```

Do you think this would make sense to include in this library?

If so, would you prefer to use an array as args or just spread the args like [this example](https://www.typescriptlang.org/play?#code/GYVwdgxgLglg9mABBOBbADnAzgUwBQB0R6ATnOjibDlgFyJ4DWOAnnMIgMIDKvULAGxwARHBAEBDEhNgIAlAG0AunMQBvAFCJtiEjiggSSPADcJAkDnpYoJGGADmqgLwA+RKXKVqWAnoAmIBD4eMD25gA0yFhYAApk6C7umjqpuvqGSClpOUQEYWCRWjlpChAx8eRK9GYWOMU5AL4NiI1Rao1yANwazRooYDaIWDAAXjiIzshomLh4AEQA7jD+UAAW81HzazgwDmtQ83Ia-QhYcEIEAnAOeCPjCwDM6AAeR8dAA)?

### PR DESCRIPTION
**Pull request by braposo**
Hey @hauptrolle!

So I've implemented the `composeUtil` function as mentioned in #1.

As part of that work I've also added tests and updated all the other utils to use this new factory function. 

It now has 100% coverage so feel free to add some additional badges / integration with something like codecov.

I removed the dev dependency of `@stitches/react` as it didn't seem to be used anywhere, but added `@stitches/core` for the types and tests. Was there a reason for including the react package?

Speaking of types, I tried my best to type the util function and it seems to be working correctly [in the playground](https://www.typescriptlang.org/play?target=7#code/JYWwDg9gTgLgBAbzgFQMIQHYDNgHMA0cAkqgM6kAKUEYyEdA1gKYaFU1OzBOmHIBKTAMYBXKKWAA3JmVJwAvnCzUQcAEQABUjGAwhACx4B6IdCZqA3ACgYATzBMU9pgBMAagEMANiKYAeZDYAPjgAXkQ4AG0AaThgDDhmWwgsYll2WnoIZgwAoIBdAC44WKYADxgWFzkKOAB+NPIMukYWPJj8uGKMJmkoBUiklMbKakzW3OQCqytyyFg4UwxtRYhwCFImAFUdLzC4P1ryyoxqxKZk1IzOHR4ggAoAOmewMZvuUmKKSPyASjCQn4rHAQSg4McqnI0JgcLgrA9gaCAPrFZBWf6hEL3STeXyo5zuXH+QJwCghAA+pLeXB4kQof1RglE4ikMnIeQBiNBoNeHBppEeUFcIiETHuXO5kvuOAw3kIQiaYwxWIQEsl6rgz0eMrlao13MiCtGNCKcBxPiY+D16vkvyt+u5CHk1rgv2sViWKxAZX2pnWmx2wC89zUIA8UFw8QAMkwsDA1IRQ+HIxh+Hh9PHfvckDBsixPhFSGAPKLikg1AASUgARjUxTUAFYwGU1Ap5RAvNAC+WK0Ja-WhS5W4oR26Zt6Q1Xa-8jEZFoYhAw4DBDHBSCJcLgeDpMHJDEKrEA).

Let me know what you think of this!

Fixes #1

### CODE REVIEW

File: src/utils/composeUtil.test.ts [hunk 1]
```diff
@@ -0,0 +1,60 @@
+import { composeUtil } from "./composeUtil";
+import { testConfig } from "./testUtils";
+import { createCss, createTokens, IAtom } from "@stitches/core";
```
Comment (hauptrolle) [inline src/utils/composeUtil.test.ts]: I would prefer to import external dependencies at the top :)

**Comment by hauptrolle** [comment]
Hey @braposo 👋 

WOW! Thank you so much for that awesome pull request! I appreciate it very much!

This looks pretty solid for me! I have just one little comment regarding the import order. Otherwise its fine and i will merge it and release a new version.

Regarding the question about `@stitches/react`: I think this was just a leftover from some tests on my side. So it can be removed.

Thank you again so much for your participation at this library! 🙇‍♂️

File: src/utils/composeUtil.test.ts (hunk 1 above)
Comment (braposo) [reply src/utils/composeUtil.test.ts]: sure, do you want to add eslint to enforce this kind of rules? I didn't want to set it up as part of this work but you can have stuff like https://github.com/benmosher/eslint-plugin-import/blob/master/docs/rules/first.md to save time in future PRs.

I've done the change in [`a0e5dc1` (#2)](https://github.com/hauptrolle/stitches-utils/pull/2/commits/a0e5dc12f39096b6e43abf16cd6a2d0cdbc86c0c)

**Comment by braposo** [comment]
Thanks for your comments @hauptrolle, it's all taken care of now. It seems that this might need to be changed very soon based on https://twitter.com/peduarte/status/1304151510163562496 but the good thing is that you only need to change this in one place!

File: src/utils/composeUtil.test.ts (hunk 1 above)
Comment (hauptrolle) [reply src/utils/composeUtil.test.ts]: Perfect! :) Yes, I created an [issue](https://github.com/hauptrolle/stitches-utils/issues/4) to keep track of adding eslint.

**Comment by hauptrolle** [comment]
@all-contributors please add @braposo  for code, doc, example, test and ideas

**Pull request merged**